### PR TITLE
feat(rappi): delivery-address management with active-address overlay

### PR DESCRIPTION
**Pull request by bobeglz**
Adds persistent delivery-location state to the rappi CLI: save named addresses (`address save casa --city ciudad-de-mexico --lat 19.36 --lng -99.17`), persist the active one (`address use casa`), and every city-bearing command resolves `--city`/`--lat`/`--lng` from it — `restaurants near --fetch-detail` with no geo flags answers for your saved address.

## Design highlights

- **Precedence:** explicit flags > `--profile` values > active address > defaults/centroid (all-or-nothing overlay in `PersistentPreRunE`; profile detection accounts for `ApplyProfileToFlags` not marking `Changed`)
- **Sentinel discipline:** nullable lat/lng + save-time `(0,0)` rejection (the `restaurants near` zero-sentinel can never be stored as data); malformed vs unknown city slugs are distinct errors (unknown-but-valid slugs like `torreon` warn-but-allow, escalated when coordinate-less)
- **Agent-native contracts:** `address` subtree is `mcp:hidden` (profile precedent); not-found → documented exit 3; the overlay note rides stderr in all modes including `--agent`; `agent-context` emits `available_addresses`/`active_address` so priming agents don't defeat the overlay with explicit `--city`
- **Generated-tree discipline:** 4 PATCH-marked one-line edits (root.go ×2, which.go, agent_context.go), all cataloged in `.printing-press-patches.json` — ledger **unioned** with the existing publish-side `review-feedback-hardening` entry, nothing erased
- Also lands the previously-unpublished 2026-06-01 hand-patches (search type-filter fix, `--agent` compact fields, snapshot item expansion)
- Generator-shaped follow-up filed as `deferred_to_upstream: active-location-context` (an active named location/context mechanism for geo-keyed pp CLIs belongs in the machine)

## Validation (all green on this branch)

- `go build ./...` · `go vet ./...` · `go test ./...` (full suite incl. 12-case overlay decision-matrix, 13-case validation table, command-level address tests)
- `gofmt` clean · `.github/scripts/verify-skill/verify_skill.py` all checks pass
- Isolated-HOME smokes: save/use/current round-trip, exit-3 not-found, `which "change my delivery address"`, `agent-context` address fields, `restaurants near --dry-run`
- PII scrub clean · no binaries staged

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by bobeglz** [comment]
@tmchow took another pass — this is ready for re-review.

**Merge conflict (resolved).** The only conflict was a modify/delete on `library/food-and-dining/rappi/.printing-press-patches.json`: `main` had migrated rappi to the `.printing-press-patches/` directory form. I resolved it by running the repo's own `.github/scripts/normalize-patches/normalize.py` against the branch, so the legacy single-array file becomes per-patch `schema_version: 2` files — de-duping the already-migrated `review-feedback-hardening` entry and routing `deferred_to_upstream` into `_meta.json`. `normalize.py --check` reports no further change, so the post-merge normalizer will see a zero diff.

**Review feedback (all addressed, with tests):**
- **P1 — `address show` nil deref** (`internal/cli/address.go`): text-mode display dereferenced `*addr.Longitude` under an `addr.Latitude != nil` guard, so a hand-edited `addresses.json` with only one coordinate would panic. Now guarded on both pointers, with a regression test.
- **Snapshot metadata loss** (`internal/cli/search.go`): `expandSnapshotItems` now stamps each expanded item with its snapshot's parent provenance (`city`/`category`/`taken_at`/…) under a `_snapshot` key, non-destructively (never clobbers an item's own fields). New `search_test.go` covers it, plus a store-level `SearchByType` test.
- **Fixed `.tmp` write race** (`internal/cli/address_store.go`): `saveAddressStore` now writes via `os.CreateTemp` (unique name) + rename, so concurrent invocations no longer race over a shared temp file. New concurrent-writers test.

CI is green and Greptile is back to passing on the current head (`83393d84d`).

One heads-up for transparency: a fresh review also surfaced a few **pre-existing, non-blocking** edge cases that this PR didn't introduce — multi-word search queries can over-filter when they hit a snapshot row, `--limit` is applied after snapshot expansion, and the sibling profile store still uses a fixed temp name. I left those out to keep this PR scoped to the conflict + review feedback; happy to take them in a follow-up if you'd like.

File: library/food-and-dining/rappi/internal/cli/address.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+			}
+			if flags.asJSON {
+				return printJSONFiltered(cmd.OutOrStdout(), addr, flags)
+			}
+			fmt.Fprintf(cmd.OutOrStdout(), "address %q:\n", addr.Label)
+			fmt.Fprintf(cmd.OutOrStdout(), "  city: %s\n", addr.City)
+			if addr.Latitude != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "  lat: %.12g\n", *addr.Latitude)
+				fmt.Fprintf(cmd.OutOrStdout(), "  lng: %.12g\n", *addr.Longitude)
+			}
```
[the remark this answers is not in the document]
Comment (bobeglz) [reply library/food-and-dining/rappi/internal/cli/address.go]: Fixed in 83393d8: the text-mode `address show` guard now checks both pointers (`addr.Latitude != nil && addr.Longitude != nil`) before dereferencing, so a hand-edited `addresses.json` with only one coordinate set no longer panics. Added a regression test covering a lat-only store. Resolving.

**Comment by tmchow** [comment]
Thanks for the PR. Greptile is still at **4/5**, which means there is at least one issue to address before this can move to maintainer review.

Please fix the Greptile finding, push an update, and let Greptile re-review the current head commit. Once Greptile reaches **5/5** and CI/dogfood evidence are clean, this can be re-triaged.

**Comment by tmchow** [comment]
Following up on the previous triage note. This PR is still waiting on author action for the same blocker(s) after 8 days:

- Greptile score is 4/5; 5/5 is required for maintainer-ready triage

Please push a fix or reply with context. If there is no author update within 14 days of the original triage request, this may be closed as stale.

**Comment by tmchow** [comment]
Quiet for a few weeks since the last ask, so I am closing this out for now. Feel free to open a fresh PR via `/printing-press publish` when you have it green again (manuscript proofs under `.manuscripts/.../proofs/` are what we look for).
